src/utils/request.ts

```diff
@@ -63,6 +57,17 @@ export function parseConditionalHeaders(headers: Headers): ConditionalHeaders {
   };
 }
 
+function getDateFromHeader(dateString: string | null): Date | undefined {
+  if (dateString === null) {
+    return undefined;
+  }
+
+  const date = new Date(dateString);
+
+  // @ts-expect-error date to number conversion
+  return date instanceof Date && !isNaN(date) ? date : undefined;
```

Comment:
```suggestion
  const date = new Date(dateString);
  return !isNaN(date.getTime()) ? date : undefined;
```
To avoid the TS err + instanceof check should be redundant now